File: Go/examples/copy_object.go
package cos

import (
	"bytes"
	"context"
	"errors"
	"fmt"
	"io/ioutil"
	"net/http"
	"net/url"
	"os"
	"strings"
	"testing"
	"time"

	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/suite"
	"github.com/tencentyun/cos-go-sdk-v5"
)

type CosTestSuite struct {
	suite.Suite
	VariableThatShouldStartAtFive int

	// CI client
	Client *cos.Client

	// Copy source client
	CClient *cos.Client

	// test_object
	TestObject string

	// special_file_name
	SepFileName string
}

var (
	UploadID string
	PartETag string
)

// 复制对象时保留对象属性
func (s *CosTestSuite) copyObject() {
  client := s.Client
  //.cssg-snippet-body-start:[copy-object]
  name := "exampleobject"
  // 上传源对象
  f := strings.NewReader("test")
  _, err := client.Object.Put(context.Background(), name, f, nil)
  assert.Nil(s.T(), err, "Test Failed")
  
  sourceURL := fmt.Sprintf("%s/%s", client.BaseURL.BucketURL.Host, name)
  dest := "example_dest"
  // 如果不是必要操作，建议上传文件时不要给单个文件设置权限，避免达到限制。若不设置默认继承桶的权限。
  // opt := &cos.ObjectCopyOptions{}
  _, _, err = client.Object.Copy(context.Background(), dest, sourceURL, nil)
  if err != nil {
      panic(err)
  }
  
  //.cssg-snippet-body-end
}

//.cssg-methods-pragma


func TestCOSTestSuite(t *testing.T) {
	suite.Run(t, new(CosTestSuite))
}

func (s *CosTestSuite) TestCopyObject() {
	// 将 examplebucket-1250000000 和 ap-guangzhou 修改为真实的信息
	u, _ := url.Parse("https://examplebucket-1250000000.cos.ap-guangzhou.myqcloud.com")
	b := &cos.BaseURL{BucketURL: u}
	c := cos.NewClient(b, &http.Client{
		Transport: &cos.AuthorizationTransport{
			SecretID:  os.Getenv("COS_KEY"),
			SecretKey: os.Getenv("COS_SECRET"),
		},
	})
  s.Client = c

	// 复制对象时保留对象属性
	s.copyObject()

	//.cssg-methods-pragma
}

File: Python/examples/copy_object.py
# -*- coding=utf-8

from qcloud_cos import CosConfig
from qcloud_cos import CosS3Client
from qcloud_cos import CosServiceError
from qcloud_cos import CosClientError

secret_id = 'COS_SECRETID'     # 替换为用户的secret_id
secret_key = 'COS_SECRETKEY'     # 替换为用户的secret_key
region = 'COS_REGION'    # 替换为用户的region
token = None               # 使用临时密钥需要传入Token，默认为空,可不填
config = CosConfig(Region=region, SecretId=secret_id, SecretKey=secret_key, Token=token)  # 获取配置对象
client = CosS3Client(config)

# 复制对象时保留对象属性
def copy_object():
    #.cssg-snippet-body-start:[copy-object]
    response = client.copy_object(
        Bucket='examplebucket-1250000000',
        Key='exampleobject',
        CopySource={
            'Bucket': 'sourcebucket-1250000000', 
            'Key': 'exampleobject', 
            'Region': 'ap-guangzhou'
        }
    )
    
    #.cssg-snippet-body-end

#.cssg-methods-pragma


# 复制对象时保留对象属性
copy_object()

#.cssg-methods-pragma